### DISCUSSION TITLE
how do i make one staff in musescore have different instruments in different voices?

### DISCUSSION THREAD

**Discussion by tperm99** [musescore/MuseScore#23444] [Q&A]
I know there was a way to do this in muse score 3 which was done by using a method through staff text, but i was wondering if there was a modern equivalent.

if there is no way to do this in muse score 4, I will be very sad and cry in the corner :(
also, they need to add that back there was no reason that it should have been deleted

**Comment by MarcSabatella** [comment]
There is not currently a way.

If you'd like to formally request this feature be designed and implemented, you can open a feature request issue.  Note the way it worked in MU3 was almost impossible to discover; it would almost certainly need a much better design.  So you might want to discuss this first on the feature request forum over at musescore.org to brainstorm ideas as to how it might work better.  Also be sure to explain your use case - and ask others discussing the idea to do the same - to be sure that any proposed design actually covers them well, and makes the most common of these case as easy as possible.

**Comment by RhinoHaggis** [comment]
I think at the most basic level the problem already solves itself on the UI side: we can already define separate voices on a staff and assign them their own phrasing, articulation and dynamics.  If these can’t already be split into separate midi channels, they should be.  Perhaps by default on any instrument that can only play one tone at a time.  Likewise notation already has a « change instrument » style.  If it’s just cosmetic it doesn’t have to be.

I don’t know what the OP’s usage case is but these two features:
1. Distinct voices can be assigned to distinct midi channels
2. The playback instrument is not fixed, but can be reassigned

with the addition of two others
3. Multiple instruments can be assigned to a single voice (handles however you want under the hood, for the additional channels)
4. The style of « change instrument » altered to permit free _cosmetic_ editing (size, font, italics/bold, rewrite the text, etc), with the actual parameters (the designated instruments) editable separately.

would cover a wide range of usage cases.

**Reply by RhinoHaggis** [reply]
Consider two examples:
1. Duet written on a single staff for both performers.

2. piano-conductor scores.

No. 2 is a good test of usage coverage. A typical P/C score is written with one grand staffs + one or two bracketed parts on which the various instruments are arranged, giving the appearance (and rehearsal function) of two pianos.  Even when the lower staff is what the pianist intends to play, it can still be peppered with both cues and instrument designations showing what instruments (if any) the piano is doubling or replacing.
The upper staff is Everything Else, also often reduced sufficiently to be playable  at the piano, but the instruments represented are constantly changing.


Here is a page from the P/C score for Hot Honey Rag from _Chicago_
![Hot Honey](https://github.com/user-attachments/assets/18183a16-892e-4ae9-b1ed-5b6b49b20bc6)
[Hot Honey Rag - Piano Conductor Score -excrpt.mscz.zip](https://github.com/user-attachments/files/17154452/Hot.Honey.Rag.-.Piano.Conductor.Score.-excrpt.mscz.zip)

There would certainly be issues.  Changing to percussion on "whatever staff line is available" would probably be out.  For existing scores, the quality of the arrangement for _playback_ would be highly variable depending on the intention of the arranger.  _Chicago_ is verbose, reproducing complete parts.  In other scores the principal function of everything not-played-by-the-piano is just to notate instrument entrances, who's-playing-what, and critical moving lines. Playback as an orchestral _arrangement_ would be very silly.

However, for new, user-created scores this could be a great idea for MuseScore, as it would let people create and share orchestral arrangements in a familiar and compact setting rather than having to dive into full conductor scores with parts.

**Reply by Fourniture** [reply]
Another case for this would also be in the case of changing stops on an organ. MuseHub does have some organs sampled which you can change the stops on, but if someone needs a sound that doesn't exist on those (e.g. zimbelstern, bells, bassoon) this would be incredibly useful. (Note that if they use "change instrument" text it changes the whole thing: both staves and pedal.

**Comment by RhinoHaggis** [comment]
Another issue would be that many instrument "doublings" are not true doublings.  "Saxes + tpts + trombone" written as block chords certainly does NOT mean that all instruments are playing all notes.  More voices would be needed than it appears at first glance, but there's not reason these problems couldn't be left up to the user to resolve (e.g. stemless noteheads for some voices, to keep the block chord look and avoid a stem war), or at least until some other method becomes available (like instrument assignment to specific noteheads within a voice).

**Comment by sammik** [comment]
see #10128 for some context
